cloud/scope/powervs_cluster.go

```diff
@@ -150,6 +150,9 @@ func NewPowerVSClusterScope(params PowerVSClusterScopeParams) (scope *PowerVSClu
 
 // PatchObject persists the cluster configuration and status.
 func (s *PowerVSClusterScope) PatchObject() error {
+	if s == nil {
```

Comment:
are you sure it is for the `s` ? may be it is for the patchHelper field!

Reply:
@mkumatag yes `s` obj is nil
```
code:
fmt.Println("obj:--- ", s)

log:
[manager] obj:---  <nil>
```

Reply:
but from this is getting triggered, we need to block in the calling method.

Reply:
I think this is being called in the defer method, lets handle it there, e.g: https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/983675a6acf3ff5f3d3bf7dbf94d40a86d268939/controllers/ibmpowervsmachine_controller.go#L142

Reply:
Yes fixed it.

Reply:
but we need to handle such situation in every defer calls

Reply:
@mkumatag you meant as a preventative measure right (since we have not came across that issue in other defer calls but yes I agree there is always a possibility)

Reply:
Updated.